### ISSUE TITLE
Issue with picking up noqa issues after other comment

### ISSUE THREAD

**Issue by jonyscathe** [plinss/flake8-noqa#10]
Some weird behaviour when both flake8-noqa and flake8-bandit are installed, with a few of the flake8-noqa codes not being raised:

- If I have a blanket noqa on a line with no violations NQA101 does not get flagged.
- If I have a noqa with a made up random code (i.e., `# noqa: ABC123`) then NQA102 does get flagged.
- But if I have a noqa with a genuine code (i.e., `# noqa: S101` or `# noqa: PD001`) then NQA102 does not get flagged.

Obviously I have no idea if the issue lies within flake8-bandit or flake8-noqa, but there does seem to be some compatibility issue there.

Running flake8==3.9.2
I am running flake-noqa==1.2.3 (though issue still exists with 1.2.2, 1.2.0, 1.1.0, 1.0.0)
I am running flake8-bandit==3.0.0 (problem doesn't seem to happen with 2.0.0 and does with 2.0.1)
The fact the problem seems to appear with flake8-bandit 2.0.1 would lead me to believe the issue is with flake8-bandit, but the only change in that project from 2.0.0 -> 2.0.1 was some setup.py changes which I wouldn't have thought would cause any issues like this: https://github.com/tylerwince/flake8-bandit/compare/v2.0.0...v2.0.1

I am also raising an issue in flake8-bandit about this given that I'm not sure which project is actually causing the problem

**Comment by plinss** [comment]
I'm not able to reproduce this, can you post some sample code?

Also, please post the output of `flake8 --version`

**Comment by jonyscathe** [comment]
Sorry, I thought this was simpler than it is....
In some simple test cases I don't get the issue - will try come up with a minimal reproduction of some sort.

**Comment by jonyscathe** [comment]
Oh I am an idiot

flake8-bandit was causing a crash due to a compatibility issue with bandit which was causing errors to not show up! So nothing at all to do with flake8-bandit...

I found this "issue" while trying to track down why NQA102 wasn't coming up in some code where I knew the issue behind the original violation had been solved.

The issue we are actually having is that if we have some line with both a mypy error and a flake8 error we have to have both a type ignore and a noqa:
```
from pandas.core.frame import DataFrame
from pandas.core.series import Series

def load_data() -> tuple[Series, DataFrame, DataFrame]:  # type: ignore[type-arg] # noqa: FNE004
```

That all works fine. Then if we change the function name so that FNE004 isn't an issue anymore:
```
from pandas.core.frame import DataFrame
from pandas.core.series import Series

def load_data_from_file() -> tuple[Series, DataFrame, DataFrame]:  # type: ignore[type-arg] # noqa: FNE004
```
Then we do not get any NQA102 errors.

So basically:
```
def load_data() -> tuple[Series, DataFrame, DataFrame]:  # type: ignore[type-arg] # noqa: FNE004
```
gives no errors correctly
```
def load_data_from_file() -> tuple[Series, DataFrame, DataFrame]:  # type: ignore[type-arg] # noqa: FNE004
```
gives no errors, but should give NQA102
```
def load_data_from_file() -> tuple[Series, DataFrame, DataFrame]:  # type: ignore[type-arg]
```
Raises a FNE004
```
def load_data_from_file() -> tuple[Series, DataFrame, DataFrame]:  # noqa: FNE004
```
Raises an NQA102 (and a mypy error)

I think it is just an issue where flake8-noqa doesn't recognise the noqa comment if it is after any other comment.
And the only possible way to suppress a mypy and flake8 error on the same line is to use the style `# type: ignore[code] # noqa: code`
